provisioner/k8s/README.md

```diff
@@ -1,3 +1,112 @@
-# k8s
+# k8s Provisioner
```

Comment:
non-blocking question: I've been thinking about this for a bit now, but does it still make sense to call this the `k8s` provisioner? Personally it feels more like the `default` or `plain` nomenclature makes more sense.

Reply:
Right now there is some inconsistency between k8s and plain in the docs. I personally don't like "plain" or "k8s" that much. We need to iron out our naming convention for this provisioner and the plain bundle spec.

Reply:
Agreed, I don't feel that this is blocking. We can address it later but I think it is worth getting a clear name for it.

Reply:
I think we need a better name than k8s for sure. Both registryv1 and this provisioner (and probably almost all others) are deploying to k8s. The name should more clearly reflect what the source content is.

Reply:
Do we need a ticket that tracks that want/need for a more descriptive plain bundle provisioner?